### ISSUE TITLE
Rule `jsx-key` fails when using spread operator

### ISSUE THREAD

**Issue by PierreGUI** [palantir/tslint-react#122]
Sometimes I like to squeeze my props and use the spread operator to pass then from parent to children components.
Not that it matters a lot but here I found the rule failing:

```tsx
const List: React.SFC<ListProps> = ({list}) => (
    <div>{list.map((elt, key) => <Child {...{elt, key}} />)}</div>
  )
}
```
Will throw `Missing "key" prop for element. (jsx-key)`

Thanks.

**Comment by onzag** [comment]
I have the same issue this issue should be reopened.

The key property is missed if it comes within an interface of its own.

```
export interface ISearchResult {
  type: string;
  id: number;
}
```

```
interface ISearchResultWithPopulateData extends ISearchResult {
  providerProps: {
    key: string;
    forId: number;
    itemDefinition: string;
    optimize: {
      onlyIncludeProperties?: string[],
      onlyIncludeIncludes?: string[],
      excludePolicies?: boolean,
      cleanOnDismount?: boolean,
    }
  };
}
```

When passing it to my Provider as

```
loader.searchResults.map((result) => {
  return (
      <ItemDefinitionProvider
        {...result.providerProps}
      >
                      
      </ItemDefinitionProvider>
    );
})
```

I get the same issue of missing key property.

**Comment by onzag** [comment]
![image](https://user-images.githubusercontent.com/10426555/72136342-43dc3180-3391-11ea-82e0-0ba1baa6cc85.png)

**Comment by adidahiya** [comment]
This library is being deprecated, see https://github.com/palantir/tslint-react/issues/210
